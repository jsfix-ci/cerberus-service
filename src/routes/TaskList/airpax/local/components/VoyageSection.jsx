import React from 'react';
import { ICON } from '../../../../../utils/constants';
import { MovementUtil } from '../../../../../utils';

const VoyageSection = ({ refDataAirlineCodes,
  journey,
  flight,
  arrivalTime,
  description,
  movementType,
  movementStatus }) => {
  const renderModeSection = () => {
    return (
      <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
        <i className={`icon-position--left ${ICON.INDIVIDUAL}`} />
        <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
          {description}
        </p>
        <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
          <span className="govuk-font-weight-bold">
            {movementType} {movementStatus}
          </span>
        </span>
      </div>
    );
  };

  const renderVoyageSection = () => {
    return (
      <div className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right">
        <i className="c-icon-aircraft" />
        <p className="content-line-one govuk-!-padding-right-2">
          {`${MovementUtil.airlineName(MovementUtil.airlineOperator(flight), refDataAirlineCodes)}, 
        flight ${MovementUtil.flightNumber(flight)}, 
        ${MovementUtil.voyageText(arrivalTime)}`}
        </p>
        <p className="govuk-body-s content-line-two govuk-!-padding-right-2">
          <span className="govuk-!-font-weight-bold">{MovementUtil.flightNumber(flight)}</span>
          <span className="dot" />
          {`${MovementUtil.formatDepartureTime(journey)}`}
          <span className="dot" />
          <span className="govuk-!-font-weight-bold">{MovementUtil.departureLoc(journey)}</span> &#8594;
          <span className="govuk-!-font-weight-bold"> {MovementUtil.arrivalLoc(journey)}</span>
          <span className="dot" />
          {MovementUtil.formatArrivalTime(journey)}
        </p>
      </div>
    );
  };

  return (
    <>
      {renderModeSection()}
      {renderVoyageSection()}
    </>
  );
};

export default VoyageSection;
