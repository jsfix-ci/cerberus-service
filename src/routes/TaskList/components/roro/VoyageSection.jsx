import React from 'react';
import { MovementUtil, VehicleUtil, VesselUtil } from '../../../../utils';
import { DESCRIPTION_MAPPING, ICON, ICON_MAPPING, MOVEMENT_MODES } from '../../../../utils/constants';
import TrailerUtil from '../../../../utils/Trailer/trailerUtil';

const VoyageSection = ({ mode,
  journey,
  vessel,
  vehicle,
  trailer,
  iconDescription,
  arrivalTime,
  totalPersons }) => {
  const getMovementModeTypeContent = () => {
    const iconFromDescription = ICON_MAPPING[mode]?.[iconDescription];

    if (iconFromDescription === ICON.CAR) {
      const vehicleReg = VehicleUtil.registration(vehicle);
      return vehicleReg || '\xa0';
    }
    if (iconFromDescription === ICON.INDIVIDUAL) {
      return '1 foot passenger';
    }
    return totalPersons ? `${totalPersons} foot passengers` : '0';
  };

  const renderModeSection = () => {
    if (mode === MOVEMENT_MODES.ACCOMPANIED_FREIGHT) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[iconDescription]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
            {VehicleUtil.make(vehicle)} {VehicleUtil.model(vehicle)}
          </p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {TrailerUtil.registration(trailer)}
            </span>
          </span>
        </div>
      );
    }
    if (mode === MOVEMENT_MODES.UNACCOMPANIED_FREIGHT) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[iconDescription]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">{'\xa0'}</p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {VehicleUtil.registration(vehicle)}
            </span>
          </span>
        </div>
      );
    }
    if (mode === MOVEMENT_MODES.TOURIST) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[iconDescription]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
            {DESCRIPTION_MAPPING[iconDescription]}
          </p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {getMovementModeTypeContent()}
            </span>
          </span>
        </div>
      );
    }
  };

  const renderVoyageSection = () => {
    return (
      <div className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right">
        <i className="c-icon-ship" />
        <p className="content-line-one govuk-!-padding-right-2">
          {`${VesselUtil.operator(vessel)} voyage of ${VesselUtil.name(vessel)}, ${MovementUtil.voyageText(arrivalTime)}`}
        </p>
        <p className="govuk-body-s content-line-two govuk-!-padding-right-2">
          {`${MovementUtil.formatDepartureTime(journey)}`}
          <span className="dot" />
          <span className="govuk-!-font-weight-bold">{MovementUtil.departureLoc(journey)}</span> &#8594;
          <span className="govuk-!-font-weight-bold"> {MovementUtil.arrivalLoc(journey)}</span>
          <span className="dot" />
          {MovementUtil.formatArrivalTime(journey)}
        </p>
      </div>
    );
  };

  return (
    <>
      {renderModeSection()}
      {renderVoyageSection()}
    </>
  );
};

export default VoyageSection;
