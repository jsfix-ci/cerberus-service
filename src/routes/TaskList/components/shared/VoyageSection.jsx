import React from 'react';
import getVoyageComponent from '../../helper/voyageSection';

import { MovementUtil } from '../../../../utils';

const VoyageSection = ({ targetTask, refDataAirlineCodes }) => {
  const mode = MovementUtil.movementMode(targetTask);
  return (
    <section className="task-list--voyage-section">
      <div>
        <div className="govuk-grid-row grid-background--greyed">
          {getVoyageComponent(mode, targetTask, refDataAirlineCodes)}
        </div>
      </div>
    </section>
  );
};

export default VoyageSection;
