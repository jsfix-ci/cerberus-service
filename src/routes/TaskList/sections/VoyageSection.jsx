import React from 'react';
import { DESCRIPTION_MAPPING, ICON, ICON_MAPPING, MOVEMENT_MODES, VIEW } from '../../../utils/constants';
import { MovementUtil, PersonUtil, VehicleUtil, VesselUtil } from '../../../utils';
import TrailerUtil from '../../../utils/Trailer/trailerUtil';

const AirpaxVoyage = ({ targetTask, refDataAirlineCodes }) => {
  const renderModeSection = () => {
    return (
      <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
        <i className={`icon-position--left ${ICON.INDIVIDUAL}`} />
        <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
          {MovementUtil.description(targetTask)}
        </p>
        <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
          <span className="govuk-font-weight-bold">
            {MovementUtil.movementType(targetTask)} {MovementUtil.status(targetTask)}
          </span>
        </span>
      </div>
    );
  };

  const renderVoyageSection = () => {
    const journey = MovementUtil.movementJourney(targetTask);
    const flight = MovementUtil.movementFlight(targetTask);
    const arrivalTime = MovementUtil.arrivalTime(journey);
    return (
      <div className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right">
        <i className="c-icon-aircraft" />
        <p className="content-line-one govuk-!-padding-right-2">
          {`${MovementUtil.airlineName(MovementUtil.airlineOperator(flight), refDataAirlineCodes)}, 
        flight ${MovementUtil.flightNumber(flight)}, 
        ${MovementUtil.voyageText(arrivalTime)}`}
        </p>
        <p className="govuk-body-s content-line-two govuk-!-padding-right-2">
          <span className="govuk-!-font-weight-bold">{MovementUtil.flightNumber(flight)}</span>
          <span className="dot" />
          {`${MovementUtil.formatDepartureTime(journey)}`}
          <span className="dot" />
          <span className="govuk-!-font-weight-bold">{MovementUtil.departureLoc(journey)}</span> &#8594;
          <span className="govuk-!-font-weight-bold"> {MovementUtil.arrivalLoc(journey)}</span>
          <span className="dot" />
          {MovementUtil.formatArrivalTime(journey)}
        </p>
      </div>
    );
  };

  return (
    <>
      {renderModeSection()}
      {renderVoyageSection()}
    </>
  );
};

const RoRoVoyage = ({ targetTask }) => {
  const getMovementModeTypeContent = (description, _mode, _targetTask) => {
    const vehicle = VehicleUtil.get(_targetTask);
    const totalPersons = PersonUtil.totalPersons(_targetTask);
    const iconFromDescription = ICON_MAPPING[_mode]?.[description];

    if (iconFromDescription === ICON.CAR) {
      const vehicleReg = VehicleUtil.vehicleReg(vehicle);
      return vehicleReg || '\xa0';
    }
    if (iconFromDescription === ICON.INDIVIDUAL) {
      return '1 foot passenger';
    }
    return totalPersons ? `${totalPersons} foot passengers` : '0';
  };

  const renderModeSection = () => {
    const mode = MovementUtil.movementMode(targetTask);
    const description = MovementUtil.iconDescription(targetTask);
    const vehicle = VehicleUtil.get(targetTask);
    const trailer = TrailerUtil.get(targetTask);

    if (mode === MOVEMENT_MODES.ACCOMPANIED_FREIGHT) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[description]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
            {VehicleUtil.vehicleMake(vehicle)} {VehicleUtil.vehicleModel(vehicle)}
          </p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {TrailerUtil.trailerReg(trailer)}
            </span>
          </span>
        </div>
      );
    }
    if (mode === MOVEMENT_MODES.UNACCOMPANIED_FREIGHT) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[description]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">{'\xa0'}</p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {VehicleUtil.vehicleReg(vehicle)}
            </span>
          </span>
        </div>
      );
    }
    if (mode === MOVEMENT_MODES.TOURIST) {
      return (
        <div className="govuk-grid-column-one-quarter govuk-!-padding-left-9">
          <i className={`icon-position--left ${ICON_MAPPING[mode]?.[description]}`} />
          <p className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1">
            {DESCRIPTION_MAPPING[description]}
          </p>
          <span className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1">
            <span className="govuk-font-weight-bold">
              {getMovementModeTypeContent(description, mode, targetTask)}
            </span>
          </span>
        </div>
      );
    }
  };

  const renderVoyageSection = () => {
    const vessel = VesselUtil.get(targetTask);
    const journey = MovementUtil.movementJourney(targetTask);
    const arrivalTime = MovementUtil.arrivalTime(journey);
    return (
      <div className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right">
        <i className="c-icon-ship" />
        <p className="content-line-one govuk-!-padding-right-2">
          {`${VesselUtil.operator(vessel)} voyage of ${VesselUtil.name(vessel)}, ${MovementUtil.voyageText(arrivalTime)}`}
        </p>
        <p className="govuk-body-s content-line-two govuk-!-padding-right-2">
          {`${MovementUtil.formatDepartureTime(journey)}`}
          <span className="dot" />
          <span className="govuk-!-font-weight-bold">{MovementUtil.departureLoc(journey)}</span> &#8594;
          <span className="govuk-!-font-weight-bold"> {MovementUtil.arrivalLoc(journey)}</span>
          <span className="dot" />
          {MovementUtil.formatArrivalTime(journey)}
        </p>
      </div>
    );
  };

  return (
    <>
      {renderModeSection()}
      {renderVoyageSection()}
    </>
  );
};

const VoyageSection = ({ view, targetTask, refDataAirlineCodes }) => {
  let voyageJsx;
  if (view === VIEW.AIRPAX) {
    voyageJsx = (
      <AirpaxVoyage
        targetTask={targetTask}
        refDataAirlineCodes={refDataAirlineCodes}
      />
    );
  }
  if (view === VIEW.RORO_V2) {
    voyageJsx = (
      <RoRoVoyage
        targetTask={targetTask}
      />
    );
  }
  return (
    <section className="task-list--voyage-section">
      <div>
        <div className="govuk-grid-row grid-background--greyed">
          {voyageJsx}
        </div>
      </div>
    </section>
  );
};

export default VoyageSection;
