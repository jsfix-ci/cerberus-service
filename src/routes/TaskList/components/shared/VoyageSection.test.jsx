import React from 'react';
import renderer from 'react-test-renderer';

import VoyageSection from './VoyageSection';

import mockRoRoTargetTask from '../../../../__fixtures__/taskData_RoRo_Accompanied_AssigneeCurrentUser.fixture.json';
import mockAirpaxTargetTask from '../../../../__fixtures__/taskData_AirPax_AssigneeCurrentUser.fixture.json';
import mockRoRoFootPassengerTask from '../../../../__fixtures__/taskData_RoRo_Foot_Passenger_AssigneeCurrentUser.fixture.json';

import refDataAirlineCodes from '../../../../__fixtures__/airpax-airline-codes.json';

describe('VoyageSection', () => {
  it('should render the voyage section for an airpax task', () => {
    const tree = renderer.create(
      <VoyageSection targetTask={mockAirpaxTargetTask} refDataAirlineCodes={refDataAirlineCodes} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the voyage section for a RoRo v2 accompanied task', () => {
    const tree = renderer.create(
      <VoyageSection targetTask={mockRoRoTargetTask} refDataAirlineCodes={refDataAirlineCodes} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the voyage section for a RoRo v2 foot passenger task', () => {
    const tree = renderer.create(
      <VoyageSection targetTask={mockRoRoFootPassengerTask} refDataAirlineCodes={refDataAirlineCodes} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
