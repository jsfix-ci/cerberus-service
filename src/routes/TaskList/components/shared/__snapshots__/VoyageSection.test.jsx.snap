// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoyageSection should render the voyage section for a RoRo v2 accompanied task 1`] = `
<section
  className="task-list--voyage-section"
>
  <div>
    <div
      className="govuk-grid-row grid-background--greyed"
    >
      <div
        className="govuk-grid-column-one-quarter govuk-!-padding-left-9"
      >
        <i
          className="icon-position--left c-icon-hgv"
        />
        <p
          className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1"
        >
          Scania
           
          R-Series
        </p>
        <span
          className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1"
        >
          <span
            className="govuk-font-weight-bold"
          >
            GB09NFD
          </span>
        </span>
      </div>
      <div
        className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right"
      >
        <i
          className="c-icon-ship"
        />
        <p
          className="content-line-one govuk-!-padding-right-2"
        >
          DFDS voyage of KHALEESI SEAWAYS, arrived 2 years ago
        </p>
        <p
          className="govuk-body-s content-line-two govuk-!-padding-right-2"
        >
          4 Aug 2020 at 13:10
          <span
            className="dot"
          />
          <span
            className="govuk-!-font-weight-bold"
          >
            DOV
          </span>
           →
          <span
            className="govuk-!-font-weight-bold"
          >
             
            CAL
          </span>
          <span
            className="dot"
          />
          4 Aug 2020 at 13:55
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`VoyageSection should render the voyage section for a RoRo v2 foot passenger task 1`] = `
<section
  className="task-list--voyage-section"
>
  <div>
    <div
      className="govuk-grid-row grid-background--greyed"
    >
      <div
        className="govuk-grid-column-one-quarter govuk-!-padding-left-9"
      >
        <i
          className="icon-position--left c-icon-person"
        />
        <p
          className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1"
        >
          Single passenger
        </p>
        <span
          className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1"
        >
          <span
            className="govuk-font-weight-bold"
          >
            1 foot passenger
          </span>
        </span>
      </div>
      <div
        className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right"
      >
        <i
          className="c-icon-ship"
        />
        <p
          className="content-line-one govuk-!-padding-right-2"
        >
          Stena Line voyage of Stena Edda, arrived 8 months ago
        </p>
        <p
          className="govuk-body-s content-line-two govuk-!-padding-right-2"
        >
          30 Dec 2021 at 10:30
          <span
            className="dot"
          />
          <span
            className="govuk-!-font-weight-bold"
          >
            BKD
          </span>
           →
          <span
            className="govuk-!-font-weight-bold"
          >
             
            BEL
          </span>
          <span
            className="dot"
          />
          30 Dec 2021 at 18:30
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`VoyageSection should render the voyage section for an airpax task 1`] = `
<section
  className="task-list--voyage-section"
>
  <div>
    <div
      className="govuk-grid-row grid-background--greyed"
    >
      <div
        className="govuk-grid-column-one-quarter govuk-!-padding-left-9"
      >
        <i
          className="icon-position--left c-icon-person"
        />
        <p
          className="govuk-body-s content-line-one govuk-!-margin-bottom-0 govuk-!-padding-left-1"
        >
          Single passenger
        </p>
        <span
          className="govuk-body-s govuk-!-margin-bottom-0 govuk-!-font-weight-bold govuk-!-padding-left-1"
        >
          <span
            className="govuk-font-weight-bold"
          >
            Passenger
             
            <strong
              className="hods-tag  airpax-status airpax-status__green"
            >
              DC
            </strong>
          </span>
        </span>
      </div>
      <div
        className="govuk-grid-column-three-quarters govuk-!-padding-right-7 align-right"
      >
        <i
          className="c-icon-aircraft"
        />
        <p
          className="content-line-one govuk-!-padding-right-2"
        >
          British Airways, 
        flight BA103, 
        Unknown
        </p>
        <p
          className="govuk-body-s content-line-two govuk-!-padding-right-2"
        >
          <span
            className="govuk-!-font-weight-bold"
          >
            BA103
          </span>
          <span
            className="dot"
          />
          7 Aug 2020 at 17:15
          <span
            className="dot"
          />
          <span
            className="govuk-!-font-weight-bold"
          >
            FRA
          </span>
           →
          <span
            className="govuk-!-font-weight-bold"
          >
             
            LHR
          </span>
          <span
            className="dot"
          />
          Unknown
        </p>
      </div>
    </div>
  </div>
</section>
`;
